### PR TITLE
[topi][relay] add operation tan to TVM

### PR DESCRIPTION
**Pull request by notoraptor**
Hi! This pull request adds operation tan to TVM. The main goal is to make it available in relay module.

### CODE REVIEW

File: python/tvm/relay/op/_tensor.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
 
 register_broadcast_schedule("log")
+register_broadcast_schedule("tan")
```
Comment (kevinthesun) [inline python/tvm/relay/op/_tensor.py]: Add shape function for tan as well?

**Comment by kevinthesun** [comment]
Minor comments. Also can you rebase master?

File: python/tvm/relay/op/_tensor.py (hunk 1 above)
Comment (notoraptor) [reply python/tvm/relay/op/_tensor.py]: Hi ! What do you mean by adding shape function for tan ?

**Comment by notoraptor** [comment]
@kevinthesun Hi ! Thanks!

In order to fix failed test (precision error), I am trying to update the `tan` implementation in llvm target (by replacing `sin/cos` with a direct call to llvm's `tan` functions). But it currently generate other issues, so I am working on it.

I can rebase now, but I guess test will fail again.

**Comment by notoraptor** [comment]
Rebased and updated to try to fix failing test.

File: python/tvm/relay/op/_tensor.py (hunk 1 above)
Comment (kevinthesun) [reply python/tvm/relay/op/_tensor.py]: https://github.com/apache/incubator-tvm/blob/b6893017d964f531e7f1600faaec4e3817fe89e2/python/tvm/relay/op/_tensor.py#L185-L217

**Comment by notoraptor** [comment]
Hi @kevinthesun I rebased and added shape function for tan in latest commit.

I still have a mismatch error in `tests/python/frontend/tensorflow/test_forward.py::test_forward_tan`. It's hard to debug because I cannot reproduce it on my computer. But I am still working on it.

**Comment by notoraptor** [comment]
@kevinthesun Tests passed ! I implemented tan in LLVM using `sin/cos` and I fixed CUDA implementation by using `tanf` instead of `__tanf` for float32.

**Comment by kevinthesun** [comment]
@notoraptor Thanks!

**Comment by tqchen** [comment]
NOTE: the git commit message messed up today due to github issue https://discuss.tvm.ai/t/github-issue-the-commit-author-is-wrong-since-today/5880/15 we should refrain from merging others' PR until tomorrow

**Comment by kevinthesun** [comment]
@tqchen Do we need to do something for this PR?

**Comment by tqchen** [comment]
See https://discuss.tvm.ai/t/response-to-github-author-attribution-bug/5909/2

**Comment by tqchen** [comment]
Hi @notoraptor @kevinthesun The community has reach a concensus to correct the github author attribution problem by reverting the affected PR, and ask the contributor to resend the pull request.

See background here https://discuss.tvm.ai/t/response-to-github-author-attribution-bug/5909
It will need an coordindated effort
- first @kevinthesun first revert the PR
- then @notoraptor to resent the PR to the master

Please reply to the post to see if you are interested in doing so. As a community we value every contribution and would like to make things right. Once both of you confirms, we can go ahead and expedite the process

**Comment by notoraptor** [comment]
Hi @tqchen ! OK for that.

Anyway, if it causes some problems, then we can just let things as it is, as my main goal was just to make `tan` available in relay !

Thanks !

**Comment by tqchen** [comment]
cc @kevinthesun please send the reverting PR, then @notoraptor can send the patch

**Pull request merged**